js/compy-stuff.js

```diff
@@ -11,15 +11,36 @@
  * @returns {number}
  */
 function testMandelbrot(c, maxIter = 200) {
-    let z = { re: 0, im: 0 }
-    for (let i = 0; i < maxIter; i++) {
-        complexSquare(z);
-        complexAdd(z, c);
-        if (squareComplexAbs(z) > 4) return i;
+    if (!testCardioid(c) && !testBulb(c)) {
+        let z = { re: 0, im: 0 }
+        for (let i = 0; i < maxIter; i++) {
+            complexSquare(z);
+            complexAdd(z, c);
+            if (squareComplexAbs(z) > 4) return i;
+        }
     }
     return maxIter;
 }
 
+/**
+ * @param {Complex} c
+ * @returns {boolean}
+ */
+function testCardioid(c) {
```

Comment:
@lovasoa Could you explain this bit to me? It goes right over my head.

Reply:
`testCardioid` tests whether a point c is in the central cardioid, and `testBulb` tests whether a point is in the circle of center (-1, 0) and radius 1/4. In both cases, we know that the point will be part of the mandelbrot set, and we can save some computation.

![image](https://user-images.githubusercontent.com/552629/59095209-313fdc80-8918-11e9-8c72-5310afaed012.png)

[More about this on wikipedia](https://en.wikipedia.org/wiki/Mandelbrot_set#Cardioid_/_bulb_checking)

Reply:
@lovasoa Thank you!